tracer/src/Datadog.Trace/ClrProfiler/CommonTracer.cs

```diff
@@ -0,0 +1,56 @@
+// <copyright file="CommonTracer.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler
+{
+    /// <summary>
+    /// This class contains methods implemented by both the automatic and manual tracer.
+    /// It is used for duplex communication.
+    /// </summary>
+    internal abstract class CommonTracer : ICommonTracer
+    {
+        private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor(typeof(CommonTracer));
+
+        public void LockSamplingPriority()
+        {
+            var traceContext = Tracer.Instance.ActiveScope?.Span.Context?.TraceContext;
+
+            if (traceContext != null)
+            {
+                traceContext.LockSamplingPriority(notifyDistributedTracer: false);
+            }
+        }
+
+        public int? TrySetSamplingPriority(int? samplingPriority)
+        {
+            var traceContext = Tracer.Instance.ActiveScope?.Span.Context?.TraceContext;
```

Comment:
I'm not sure if all spans in a TraceContext have the same traceId, but should we be doing some validation that the `ActiveScope.Span.Context.TraceId` matches the `TraceId` of the caller?